src/Array.mo

```diff
@@ -713,5 +713,22 @@ module {
   /// Runtime: O(1)
   ///
   /// Space: O(1)
-  public func size<X>(array : [X]) : Nat = array.size()
-}
+  public func size<X>(array : [X]) : Nat = array.size();
+
+  /// Returns a new subarray from the given array provided the start index and length of elements in the subarray
+  ///
+  /// Limitations: Traps if the start index + length is greater than the size of the array
+  ///
+  /// ```motoko include=import
+  ///
+  /// let array = [1,2,3,4,5];
+  /// let subArray = Array.subArray<Nat>(array, 2, 3);
+  /// Runtime: O(length);
+  /// Space: O(length);
+  public func subArray<X>(arr: [X], start: Nat, length: Nat): [X] {
+    if (start + length > arr.size()) { Prim.trap("Array.subArray: start index + subArray length out of bounds") };
```

Comment:
```suggestion
    if (start + length > arr.size()) { Prim.trap("Array.subArray") };
```

Reply:
Too much information ;-> This is enough for users to track down the bug, I think.

Reply:
What about "out of bounds error"? Ideally people don't have to dig into the base library code to see what happened.